### PR TITLE
[darksky] Added support for weather alerts

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added support for weather alerts
- Changed channel types for precipitation to `Number:Speed` / `MILLIMETRE_PER_HOUR` (API breaking)
- Minor code improvements

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.darksky/README.md [hunk 1]
```diff
@@ -61,9 +63,9 @@ Once the parameters `forecastHours` or `forecastDays` will be changed, the avail
 | current          | gust-speed         | Number:Speed         | Current gust speed. **Advanced**                                        |
 | current          | cloudiness         | Number:Dimensionless | Current cloudiness.                                                     |
 | current          | visibility         | Number:Length        | Current visibility.                                                     |
-| current          | rain               | Number:Length        | Rain volume of the last hour.                                           |
-| current          | snow               | Number:Length        | Snow volume of the last hour.                                           |
-| current          | precip-intensity   | Number:Length        | Current precipitation intensity.                                        |
+| current          | rain               | Number:Speed         | Rain volume of the last hour.                                           |
```
Comment (J-N-K) [inline bundles/org.openhab.binding.darksky/README.md]: Precipitation is measured in l/m2 (or more commonly mm). This is volume/area, not speed (distance/time). The [german Wikipedia article](https://de.wikipedia.org/wiki/Niederschlag#Niederschlagsmenge) also supports that . If you bring the time in, it's a flow (volume/time).

Why did you change it?

Comment (cweitkamp) [reply bundles/org.openhab.binding.darksky/README.md]: The API returns "`precipIntensity`: Millimeters per hour." (see https://darksky.net/dev/docs#forecast-request). That fits per definition fit the dimension speed. I know it sounds weird in the first moment. To make it more clear I should maybe adopt the descriptions and translations too.

Comment (J-N-K) [reply bundles/org.openhab.binding.darksky/README.md]: Being a physicist by trade I'm nitpicking here xD. In fact it's volume/(area*time), which is NOT speed, even if the dimension fits (the old `Length` was wrong in the same way). But ok.

Comment (cweitkamp) [reply bundles/org.openhab.binding.darksky/README.md]: Done.

**Comment by cweitkamp** [comment]
Thanks. I will add a notice to the [update script](https://github.com/openhab/openhab-distro/blob/eeb5ced081dc3df1e71566f09d48faf16c342957/distributions/openhab/src/main/resources/bin/update.lst#L29).
